sdk/storage/Azure.Storage.DataMovement/src/ServiceToServiceTransferJob.cs

```diff
@@ -80,31 +68,48 @@ public override async IAsyncEnumerable<JobPartInternal> ProcessJobToJobPartAsync
             JobPartStatusEvents += JobPartEvent;
             await OnJobStatusChangedAsync(StorageTransferStatus.InProgress).ConfigureAwait(false);
             int partNum = 0;
-            if (_isSingleResource)
-            {
-                // Single resource transfer, we can skip to chunking the job.
-                ServiceToServiceJobPart part = new ServiceToServiceJobPart(this, partNum);
-                _jobParts.Add(part);
-                yield return part;
-            }
-            else
+            if (_jobParts.Count == 0)
```

Comment:
very nested and not clear what's being switched on

Reply:
It checks to see whether or not it's a resumable job based on if there already prescanned job parts.

I just wrote a comment above this line to clear up the confusion.